File: python/warmup-2/string_splosion.py
# Given a non-empty string like "Code" return a string like "CCoCodCode".

# string_splosion('Code') → 'CCoCodCode'
# string_splosion('abc') → 'aababc'
# string_splosion('ab') → 'aab'

def string_splosion(str):
    result = ""
    splode = ""

    for i in range(len(str) + 1):
        result += str[0:i]

    return result

File: java/warmup-2/stringSplosion.java
// Given a non-empty string like "Code" return a string like "CCoCodCode".

// stringSplosion("Code") → "CCoCodCode"
// stringSplosion("abc") → "aababc"
// stringSplosion("ab") → "aab"

public String stringSplosion(String str) {
  String splode = "";
  String result = "";
  for (int i = 0; i < str.length() + 1; i++) {
    splode = str.substring(0, i);
    result += splode;
  }
  return result;
}
